jdaviz/configs/specviz/tests/test_helper.py

```diff
@@ -77,6 +78,66 @@ def test_get_spectra_label_redshift_warn(self):
         assert_quantity_allclose(spectra.flux,
                                  self.spec.flux, atol=1e-5*u.Unit(self.spec.flux.unit))
 
+    def test_get_spectral_regions_none(self):
+        spec_region = self.spec_app.get_spectral_regions()
+
+        assert spec_region == {}
+
+    def test_get_spectral_regions_one(self):
+        self.spec_app.app.get_viewer("spectrum-viewer").apply_roi(XRangeROI(6000, 6500))
+        spec_region = self.spec_app.get_spectral_regions()
+        assert len(spec_region['Subset 1'].subregions) == 1
+
+    def test_get_spectral_regions_two(self):
+        spectrum_viewer = self.spec_app.app.get_viewer("spectrum-viewer")
+
+        # Set the active edit_subset_mode to OrMode to be able to add multiple subregions
+        spectrum_viewer.session.edit_subset_mode._mode = OrMode
+        self.spec_app.app.get_viewer("spectrum-viewer").apply_roi(XRangeROI(6000, 6500))
+        self.spec_app.app.get_viewer("spectrum-viewer").apply_roi(XRangeROI(7300, 7800))
+
+        spec_region = self.spec_app.get_spectral_regions()
+
+        assert len(spec_region['Subset 1'].subregions) == 2
+
+    def test_get_spectral_regions_three(self):
+        spectrum_viewer = self.spec_app.app.get_viewer("spectrum-viewer")
+
+        spectrum_viewer.session.edit_subset_mode._mode = OrMode
+        self.spec_app.app.get_viewer("spectrum-viewer").apply_roi(XRangeROI(6000, 6400))
+        self.spec_app.app.get_viewer("spectrum-viewer").apply_roi(XRangeROI(6600, 7000))
+        self.spec_app.app.get_viewer("spectrum-viewer").apply_roi(XRangeROI(7300, 7800))
+
+        spec_region = self.spec_app.get_spectral_regions()
+
+        assert len(spec_region['Subset 1'].subregions) == 3
+        # Assert correct values for test with 3 subregions
+        assert_quantity_allclose(spec_region['Subset 1'].subregions[0][0].value,
+                                 6000., atol=1e-5)
+        assert_quantity_allclose(spec_region['Subset 1'].subregions[0][1].value,
+                                 6222.22222222, atol=1e-5)
```

Comment:
Shouldn't the upper bound here match the ROI you defined? (6400, maybe slightly off depending on pixel spacing?)

Reply:
No it is exact, so 6444 will not be included.

Reply:
For posterity's sake, the reason the values looked far off from what I expected is because the test spectrum only has 10 points (if I'm remembering correctly) and thus these are the values at the points nearest the specified bounds.